middleware/tokenauth/tokenauth.go

```diff
@@ -0,0 +1,151 @@
+// Package tokenauth provides jwt token authorisation middleware
+// supports HMAC, RSA, ECDSA, RSAPSS algorithms
+// uses github.com/dgrijalva/jwt-go for jwt implementation
+package tokenauth
+
+import (
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gobuffalo/envy"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/gobuffalo/buffalo"
+	"github.com/pkg/errors"
+)
+
+var (
+	// ErrTokenInvalid is returned when the token provided is invalid
+	ErrTokenInvalid = errors.New("token invalid")
+	// ErrNoToken is returned if no token is supplied in the request.
+	ErrNoToken = errors.New("token not found in request")
+	// ErrBadSigningMethod is returned if the token sign method in the request
+	// does not match the signing method used
+	ErrBadSigningMethod = errors.New("unexpected signing method")
+)
+
+// Options for the JWT middleware
+type Options struct {
+	SignMethod    jwt.SigningMethod
+	GetKey        func()(interface{}, error)
```

Comment:
This needs to take `jwt.SigningMethod` for those who want to build their own function, they might need that information.

Reply:
@markbates added this.